GVFS/GVFS.Common/NuGetUpgrader/NuGetUpgrader.cs

```diff
@@ -0,0 +1,466 @@
+
+            return success;
+        }
+
+        public bool TryRunInstaller(InstallActionWrapper installActionWrapper, out string error)
+        {
+            string localError = null;
+            int installerExitCode;
+            bool installSuccessful = true;
+            using (ITracer activity = this.tracer.StartActivity(nameof(this.TryRunInstaller), EventLevel.Informational))
+            {
+                try
+                {
+                    string platformKey = ReleaseManifest.WindowsPlatformKey;
+                    string upgradesDirectoryPath = ProductUpgraderInfo.GetUpgradesDirectoryPath();
+
+                    Exception e;
+                    if (!this.fileSystem.TryDeleteDirectory(this.localUpgradeServices.TempPath, out e))
+                    {
+                        error = e?.Message;
+                        return false;
+                    }
+
+                    string extractedPackagePath = this.UnzipPackageToTempLocation();
+                    this.releaseManifest = ReleaseManifest.FromJsonFile(Path.Combine(extractedPackagePath, "content", "install-manifest.json"));
+                    InstallManifestPlatform platformInstallManifest = this.releaseManifest.PlatformInstallManifests[platformKey];
+
+                    if (platformInstallManifest == null)
+                    {
+                        activity.RelatedError($"Extracted ReleaseManifest from JSON, but there was no entry for {platformKey}.");
+                        error = $"No entry in the manifest for the current platform ({platformKey}). Please verify the upgrade package.";
+                        return false;
+                    }
+
+                    activity.RelatedInfo($"Extracted ReleaseManifest from JSON. InstallActions: {platformInstallManifest.InstallActions.Count}");
+
+                    this.fileSystem.CreateDirectory(upgradesDirectoryPath);
+
+                    foreach (ManifestEntry entry in platformInstallManifest.InstallActions)
+                    {
+                        string installerPath = Path.Combine(extractedPackagePath, "content", entry.InstallerRelativePath);
+
+                        activity.RelatedInfo(
+                            $"Running install action: Name: {entry.Name}, Version: {entry.Version}" +
+                            $"InstallerPath: {installerPath} Args: {entry.Args}");
+
+                        installActionWrapper(
+                            () =>
+                            {
+                                if (!this.dryRun)
+                                {
+                                    this.localUpgradeServices.RunInstaller(installerPath, entry.Args, out installerExitCode, out localError);
+                                }
+                                else
+                                {
+                                    // This is a temporary workaround to force the step to be written
+                                    // out to the console.
```

Comment:
Should this temporary workaround be removed?  If not, what is the plan for removing it?

Reply:
There are two reasons:

1. Without any delay, these actions are fast...
2. The logic to display messages to the console only displays messages if the action takes > 2 seconds. So, the install step would not be printed to the console.

Before we remove this, we need a way to have ensure the messages are printed to console for each step (otherwise, it is hard to see what steps are run...). I think we might even want this assurance for installation steps in general - we would want to display every installation step to the user.

I was going to look at a way to force the message to be displayed, even if the action takes < 2s. Or maybe this is already possible, but I overlooked it.

Reply:
Isn't there an option to change the delay to `0`?